addon/decorators/track-on-constructor.ts

```diff
@@ -0,0 +1,32 @@
+import { ActivityType } from '../services/activity-tracking';
+import { assert } from '@ember/debug';
+import { getOwner } from '@ember/application';
+import { getOwnConfig } from '@embroider/macros';
+
+interface Constructable {
+  new (...args: any[]): any;
+}
+
+export function trackOnConstructor(actionDescription: string, actionType: ActivityType) {
+  return function extendClass<T extends Constructable>(BaseClass: T) {
+    return class extends BaseClass {
+      constructor(...args: any) {
+        super(...args);
+        assert(
+          '[decorator][track-on-constructor] An actionDescription needs to be passed for the activity-log to make sense.',
+          typeof actionDescription === 'string'
+        );
+        getOwner(this)
+          .lookup('service:activity-tracking')
+          .log({
```

Comment:
to be honest i am wondering if the API of the ActivityTracking service is good 🤔 ,

4 (possibly 5) of the 7 will always be the same, here and in addon/helpers/log-activity.ts.

should the log activities take only three arguments? `type` and `action` and some extra?

Activity object itself ( the one sent to the backend being created inside the service ).

PS: i figured ` getOwner(this).lookup('service:router').currentRouteName` could be called inside  the activityTracking service, but TBC